src/Libraries/RevitNodes/Elements/FamilyInstance.cs

```diff
@@ -184,6 +276,29 @@ private void InternalSetPosition(XYZ fi)
             TransactionManager.Instance.TransactionTaskDone();
         }
 
+        private void InternalSetPosition(Line pos)
+        {
+            TransactionManager.Instance.EnsureInTransaction(Document);
+
+            //var lp = InternalFamilyInstance.get_Parameter(BuiltInParameter.CURVE_IS_REFERENCE_LINE).Set(pos.Reference.ElementId);
+            var lp = InternalFamilyInstance.Location as LocationCurve;
+            lp.Curve = pos;
```

Comment:
Updating the Location may not create a host relationship. @Randy-Ma, I am correct? @aparajit-pratap can you please verify this. Create a family on curve using Dynamo and then modify that curve in Dynamo. You will see that family instance is updated. Now close Dynamo and modify the curve in Revit if the family instance is updated then we have a correct hosting relationship.